src/apps/activity_assignments/service.py

```diff
@@ -225,6 +297,99 @@ async def _get_assignments_entities(
+    ) -> list[ActivityAssignment]:
+        """
+        Unassigns multiple activity assignments by marking them as deleted.
+
+        This method takes a list of assignment requests, retrieves the corresponding
+        assignment entities from the database, marks them as deleted, and then returns
+        the updated assignments.
+
+        Parameters:
+        -----------
+        applet_id : uuid.UUID
+            The ID of the applet for which the assignments are being unassigned.
+
+        assignments_unassign : list[ActivityAssignmentCreate]
+            A list of assignment creation objects that specify which assignments to unassign.
+            Each object contains details such as `activity_id`, `activity_flow_id`,
+            `respondent_subject_id`, and `target_subject_id`.
+
+        Returns:
+        --------
+        list[ActivityAssignment]
+            A list of `ActivityAssignment` objects that have been marked as deleted.
+
+        Process:
+        --------
+        1. Retrieves all relevant entities (activities, flows, subjects) using
+        `_get_assignments_entities`.
+        2. Validates each assignment and retrieves the corresponding assignment ID
+        using `_get_assignment_by_activity_or_flow_and_subject_id`.
+        3. Creates a schema for each assignment, marking it as deleted.
+        4. Unassigns the assignments in the database using `ActivityAssigmentCRUD.unassign_many`.
+        5. Optionally, sends emails based on the respondent activities (not yet implemented).
+        6. Returns a list of the updated `ActivityAssignment` objects.
+
+        """
+        entities = await self._get_assignments_entities(applet_id, assignments_unassign)
+
+        respondent_activities: dict[uuid.UUID, list[str]] = defaultdict(list)
+        schemas = []
+        for assignment in assignments_unassign:
+            activity_or_flow_name: str = self._validate_assignment_and_get_activity_or_flow_name(assignment, entities)
+
+            id_assignment = await self._get_assignment_by_activity_or_flow_and_subject_id(assignment)
+            schema = ActivityAssigmentSchema(
+                id=id_assignment,
+                activity_id=assignment.activity_id,
+                activity_flow_id=assignment.activity_flow_id,
+                respondent_subject_id=assignment.respondent_subject_id,
+                target_subject_id=assignment.target_subject_id,
+                is_deleted=True,
+            )
+
+            if entities.respondent_subjects[assignment.respondent_subject_id].user_id:
+                respondent_activities[assignment.respondent_subject_id].append(activity_or_flow_name)
+
+            schemas.append(schema)
+
+        assignment_schemas: list[ActivityAssigmentSchema] = await ActivityAssigmentCRUD(self.session).unassign_many(
+            schemas
+        )
+
+        # Todo: send emails based on respondent_activities array
```

Comment:
This is actually related with the next comment. I leave it there because I didn't knew if we want to create the send email feature for assignments and unsassignments. I was thinking in using "entities" to retrieve user information. 

`        entities = await self._get_assignments_entities(applet_id, assignments_unassign)
...
            if entities.respondent_subjects[assignment.respondent_subject_id].user_id:
                respondent_activities[assignment.respondent_subject_id].append(activity_or_flow_name)
`

This Todo comes from the assignment feature. I will delete if it's not important, I forget to ask and then I left it.

Reply:
Ah I see! As far as I know, we're not planning to send emails during unassign. I think unless this gets added to the JIRA backlog, we don't need to include a TODO about it, but it's a good thought!